### PR TITLE
Mod kitsu implement pyblish plugins

### PR DESCRIPTION
**Pull request by ClementHector**
## Brief description
Integrate data generated with OpenPype in Kitsu

## Description
From publish integrate the following data into OpenPype:
 - [X] comment
 - [X] intent/status
 - [X] review

## Additional info
A rebase has been made from 3.9.2

### CODE REVIEW

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    order = pyblish.api.CollectorOrder + 0.499
+    label = "Kitsu entities"
+
+    def process(self, context):
+
+        asset_data = context.data["assetEntity"]["data"]
+        zoo_asset_data = asset_data.get("zou")
+        if not zoo_asset_data:
+            raise AssertionError("Zoo asset data not found in OpenPype!")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]:
```suggestion
            raise AssertionError("Zou asset data not found in OpenPype!")
```

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        self.log.debug("Collected zoo asset data: {}".format(zoo_asset_data))
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]:
```suggestion
        self.log.debug("Collected zou asset data: {}".format(zoo_asset_data))
```

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        zoo_task_data = asset_data["tasks"][os.environ["AVALON_TASK"]].get("zou")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]: Rename as `zou_task_data`

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        kitsu_project = gazu.project.get_project(zoo_asset_data["project_id"])
+        if not kitsu_project:
+            raise AssertionError("Project not not found in kitsu!")
+        context.data["kitsu_project"] = kitsu_project
+        self.log.debug("Collect kitsu project: {}".format(kitsu_project))
+
+        kitsu_asset = gazu.asset.get_asset(zoo_asset_data["id"])
+        if not kitsu_asset:
+            raise AssertionError("Asset not not found in kitsu!")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]:
```suggestion
            raise AssertionError("Asset not found in kitsu!")
```

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        context.data["kitsu_asset"] = kitsu_asset
+        self.log.debug("Collect kitsu asset: {}".format(kitsu_asset))
+
+        if zoo_task_data:
+            kitsu_task = gazu.task.get_task(zoo_task_data["id"]) 
+            if not kitsu_task:
+                raise AssertionError("Task not not found in kitsu!")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]:
```suggestion
                raise AssertionError("Task not found in kitsu!")
```

File: openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                        os.environ["AVALON_TASK"]
+                    )
+                )
+
+            kitsu_task = gazu.task.get_task_by_name(
+                kitsu_asset,
+                kitsu_task_type
+            )
+            if not kitsu_task:
+                raise AssertionError("Task not not found in kitsu!")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py]:
```suggestion
                raise AssertionError("Task not found in kitsu!")
```

File: openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    def process(self, instance):
+
+        context = instance.context
+        task = context.data["kitsu_task"]
+        comment = context.data["kitsu_comment"]
+
+        for representation in instance.data.get("representations", []):
+
+            local_path = representation.get("published_path")
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py]: Is `local_path` a movie file path? If yes, could you please explicit the name?

File: openpype/modules/kitsu/plugins/publish/validate_kitsu_intent.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    def process(self, context):
+
+        publish_status = context.data.get("intent", {}).get("value")
+        if not publish_status:
+            self.log.info("Status is not set.")
+
+        kitsu_status = gazu.task.get_task_status_by_short_name(publish_status)
+        if not kitsu_status:
+            raise AssertionError(
+                "Status `{}` not not found in kitsu!".format(kitsu_status)
```
Comment (Tilix4) [inline openpype/modules/kitsu/plugins/publish/validate_kitsu_intent.py]:
```suggestion
                "Status `{}` not found in kitsu!".format(kitsu_status)
```

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Great job, thanks a lot!

Is this code required to be Python2 compatible? If no, it'd be better to use fStrings instead of `.format`

File: openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py (hunk 7 above)
Comment (ClementHector) [reply openpype/modules/kitsu/plugins/publish/integrate_kitsu_review.py]: It is the review file path.

**Comment by ClementHector** [comment]
> Great job, thanks a lot!
> 
> Is this code required to be Python2 compatible? If no, it'd be better to use fStrings instead of `.format`

Publish is run in other software with python 2 like maya 2020 that's why I use `.format`

**Review by Tilix4 — APPROVED** [review]

**Pull request merged**